dromedary.py

```diff
@@ -0,0 +1,329 @@
+current or a given Opam switch.
+
+Examples:
+
+  To add all packages of the current Opam switch to the file ./third-party/BUCK
+    python3 dromedary.py -o ./third-party/BUCK
+
+  To add all packages of the Opam switch 'switch_name' to the file
+  ./third-party/BUCK:
+    python3 dromedary.py -s switch_name -o ./third-party/BUCK
+
+  To add all packages of the current Opam switch except ocaml-lsp-server to the
+  file ./third-party/BUCK:
+    python3 dromedary.py -e ocaml-lsp-server -o ./third-party/BUCK""",
+        epilog="For details, see https://github.com/facebook/ocaml-scripts.",
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="BUCK_FILE",
+        required=True,
+        help="Output buck file name. Mandatory.",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--switch",
+        metavar="SWITCH",
+        required=False,
+        help="Use Opam switch SWITCH instead of the current one. Optional.",
+    )
+
+    parser.add_argument(
+        "-e",
+        "--exclude",
+        nargs="*",
+        metavar="PACKAGE",
+        required=False,
+        help="A list of space separated package names to exclude from BUCK_FILE. Optional.",
+    )
+
+    parser.add_argument(
+        "-r",
+        "--root",
+        metavar="ROOT",
+        required=False,
+        default="opam",
+        help="Relative path used as a root for all paths in the generated file. Optional, default is 'opam'.",
+    )
+
+    parser.add_argument(
+        "-j",
+        "--json",
+        metavar="JSON_FILE",
+        required=False,
+        help="The name of the intermediate JSON file to generate. Optional.",
+    )
+
+    parser.add_argument(
+        "-p",
+        "--package",
+        metavar="PACKAGE",
+        required=False,
+        help="Only process this single opam package PACKAGE. Optional.",
+    )
+
+    return parser.parse_args()
+
+
+###############################################################################
+def main() -> None:
+    """Main entry point of this script."""
+    source_dir = Path(__file__).parent.resolve()
+    json_script = source_dir.joinpath(JSON_SCRIPT_NAME)
+    if not json_script.exists():
+        print(f"ERROR: script file {json_script} not found!", file=sys.stderr)
+        os.remove(INTERMEDIATE_JSON)  # always remove the temporary file.
+        sys.exit(1)
+    rules_script = source_dir.joinpath(RULE_SCRIPT_NAME)
+    if not rules_script.exists():
+        print(f"ERROR: script file {rules_script} not found!", file=sys.stderr)
+        os.remove(INTERMEDIATE_JSON)  # always remove the temporary file.
+        sys.exit(1)
+
+    args = parse_command_line()
+
+    output_file: str = args.output
+    print(f"Generating buck2 file {output_file}")
+
+    exclude_list = args.exclude
+
+    switch_name = args.switch
+    if switch_name is None:
+        switch_name = opam_switch_name()
+    opam_env = opam_switch_env(switch_name)
```

Comment:
This should be rewired a bit: you're calling `opam_switch_env` twice in a row if the name is missing. Just it call it once ?

Reply:
Oh, yes. I forgot about that when refactoring the `opam_switch_?` functions.